packages/user-service/src/helpers.ts

```diff
@@ -25,17 +28,44 @@ class UserGroupApiHelper {
       ldapClient.search(this.ldapBase, search_options, (err, response) => {
         response.on(`searchEntry`, (entry) => {
           profile = entry.object;
-        });
-        response.on(`error`, (error: Error) => {
-          console.error(`LDAP error: ` + error.message);
-        });
-        response.on(`end`, (result: any) => {
-          resolve(profile);
-        });
-      });
-    });
+        } );
+        response.on( `error`, ( error: Error ) => {
+          console.error( `LDAP error: ` + error.message );
```

Comment:
The promise should be rejected in `onError`

Reply:
Updated